impl/src/test/groovy/com/okta/jwt/impl/jjwt/JjwtAccessTokenVerifierBuilderTest.groovy

```diff
@@ -0,0 +1,148 @@
+import static org.hamcrest.Matchers.instanceOf
+import static org.hamcrest.Matchers.is
+
+class JjwtAccessTokenVerifierBuilderTest {
+
+    @Test
+    void orgIssuerTest() {
+        def verifier = new JjwtAccessTokenVerifierBuilder()
+            .setIssuer("https://issuer.example.com")
+            .setAudience("foo-aud")
+            .build()
+
+        assertThat verifier.audience, is("foo-aud")
+        assertThat verifier.issuer, is("https://issuer.example.com")
+        assertThat verifier.leeway, is(120L)
+        assertThat verifier.keyResolver, instanceOf(RemoteJwkSigningKeyResolver)
+        assertThat verifier.keyResolver.jwkUri, is(new URL("https://issuer.example.com/oauth2/v1/keys"))
+    }
+
+    @Test
+    void customIssuerTest() {
+        def verifier = new JjwtAccessTokenVerifierBuilder()
+            .setIssuer("https://issuer.example.com/oauth2/anAsId")
+            .setAudience("foo-aud")
+            .build()
+
+        assertThat verifier.audience, is("foo-aud")
+        assertThat verifier.issuer, is("https://issuer.example.com/oauth2/anAsId")
+        assertThat verifier.leeway, is(120L)
+        assertThat verifier.keyResolver, instanceOf(RemoteJwkSigningKeyResolver)
+        assertThat verifier.keyResolver.jwkUri, is(new URL("https://issuer.example.com/oauth2/anAsId/v1/keys"))
+    }
+
+    @Test(dataProvider = "validIssuers")
+    void formatIssuerTest(String issuer) {
+        def verifier = new JjwtAccessTokenVerifierBuilder()
+            .setIssuer(issuer)
+            .setAudience("foo-aud")
+            .build()
+
+        assertThat verifier.issuer, is("https://valid.example.com/oauth2/default")
+    }
+
+    @Test
+    void issuer_nullTest() {
+        TestUtil.expect IllegalArgumentException, {
+            new JjwtAccessTokenVerifierBuilder()
+                    .setIssuer(null)
+                    .setAudience("foo-aud")
+                    .build()
+        }
+    }
+
+    @Test(dataProvider = "invalidIssuers")
+    void invalidIssuersTest(String issuer) {
+        TestUtil.expect IllegalArgumentException, {
+            new JjwtAccessTokenVerifierBuilder()
+                    .setIssuer(issuer.toString())
+                    .build()
+        }
+    }
+
+    @DataProvider(name = "invalidIssuers")
+    Object[][] invalidIssuers() {
+        return [
+                [""],
+                [" "],
+                ["not a url"],
+                ["http://invalid.example.com/oauth2/default"], //not https
+        ]
+    }
+
+    @DataProvider(name = "validIssuers")
+    Object[][] validIssuers() {
+        return [
+                ["https://valid.example.com/oauth2/default"],
+                [" https://valid.example.com/oauth2/default "],
+                ["https://valid.example.com/oauth2/default/"],
+                ["https://valid.example.com/oauth2/default/ "],
+                [" https://valid.example.com/oauth2/default/ "],
+        ]
+    }
+
+    @Test
+    void audience_nullTest() {
```

Comment:
Do you have any other test to verify that we fail when an audience is empty?

Reply:
Yup, take a look at: `invalidAudienceIds` (that technique doesn't work with `null`s)